addon/components/mdc-tab-bar-scroller.js

```diff
@@ -0,0 +1,141 @@
+import Ember from 'ember';
+import { MDCTabBarScrollerFoundation } from '@material/tabs';
+import { getCorrectPropertyName } from '@material/animation';
+import { MDCComponent } from '../mixins/mdc-component';
+import layout from '../templates/components/mdc-tab-bar-scroller';
+import getElementProperty from '../utils/get-element-property';
+import getComponentProperty from '../utils/get-component-property';
+import styleComputed from '../utils/style-computed';
+
+const { get, computed, set, run } = Ember;
+const { cssClasses, strings } = MDCTabBarScrollerFoundation;
+
+export default Ember.Component.extend(MDCComponent, {
+  //region Ember Hooks
+  classNames: ['mdc-tab-bar-scroller'],
+  layout,
+  init() {
+    this._super(...arguments);
+    set(this, 'backIndicatorClasses', Ember.A([]));
+    set(this, 'forwardIndicatorClasses', Ember.A([]));
+    set(this, 'mdcScrollFrameStyles', {});
+  },
+  didInsertElement() {
+    this._super(...arguments);
+    set(this, 'scrollFrameElement', getElementProperty(this, 'querySelector')(strings.FRAME_SELECTOR));
+    set(this, 'tabBarElement', getElementProperty(this, 'querySelector')(strings.TABS_SELECTOR));
+    set(this, 'forwardIndicatorElement', getElementProperty(this, 'querySelector')(strings.INDICATOR_FORWARD_SELECTOR));
+    set(this, 'backIndicatorElement', getElementProperty(this, 'querySelector')(strings.INDICATOR_BACK_SELECTOR));
+  },
+  //endregion
+
+  //region Properties
+  /**
+   * @type {Object}
+   */
+  CLASS_NAMES: cssClasses,
+
+  /**
+   * @type {Object}
```

Comment:
I believe these should be `@property` not `@type`, and should be [`HTMLElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement) instead of `Object`.